pkg/sql/pgwire/pgwire_test.go

```diff
@@ -2311,3 +2311,35 @@ func TestCancelRequest(t *testing.T) {
 		t.Fatalf("expected 1 cancel request, got %d", count)
 	}
 }
+
+func TestFailPrepareFailsTxn(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	s, _, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer s.Stopper().Stop(context.TODO())
+
+	pgURL, cleanupFn := sqlutils.PGUrl(t, s.ServingAddr(), t.Name(), url.User(security.RootUser))
+	defer cleanupFn()
+
+	db, err := gosql.Open("postgres", pgURL.String())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	tx, err := db.Begin()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = tx.Prepare("select fail")
```

Comment:
just curious: why not `if _, err :=..` here/below?

Reply:
no reason, just didn't do it - done now